### DISCUSSION TITLE
Why do we need json files to be existed in .next directory when we build static pages

### DISCUSSION THREAD

**Discussion by thmohd** [vercel/next.js#17785] [Help] [answered]
So I am using `getStaticPaths` and `getStaticProps` to build static pages but when I look at the folder `server/static/BUILD_ID/pages/PAGE_NAME/` I see the html page and a json file page like `a.html` and `a.json` and so on?

My question, why do we need the json file if the html file has the same data under `__NEXT_DATA__`? and if I try to remove the json files I get `404`.

**Comment by lfades** [comment, marked as answer]
@thmohd For client-side navigations, the initial HTML has the data inside `__NEXT_DATA__` but that only applies for that first render, a client side navigation will populate the new page with the data in the json.

**Reply by thmohd** [reply]
Thank you. But if I land on the static generated `content/SLUG` page directly for example without using client navigation, I still get 404 if the json file is not exist. Is there a way to make this work without having the json files in order to reduce  the number of files uploaded to the server?
